auth/src/main/java/com/firebase/ui/auth/util/data/PhoneNumberUtils.java

```diff
@@ -1362,13 +183,489 @@ private static String stripPlusSign(String phoneNumber) {
+        map.put(597, singletonList("SR"));
+        map.put(598, singletonList("UY"));
+        map.put(599, asList("CW", "BQ"));
+        map.put(670, singletonList("TL"));
+        map.put(672, singletonList("NF"));
+        map.put(673, singletonList("BN"));
+        map.put(674, singletonList("NR"));
+        map.put(675, singletonList("PG"));
+        map.put(676, singletonList("TO"));
+        map.put(677, singletonList("SB"));
+        map.put(678, singletonList("VU"));
+        map.put(679, singletonList("FJ"));
+        map.put(680, singletonList("PW"));
+        map.put(681, singletonList("WF"));
+        map.put(682, singletonList("CK"));
+        map.put(683, singletonList("NU"));
+        map.put(685, singletonList("WS"));
+        map.put(686, singletonList("KI"));
+        map.put(687, singletonList("NC"));
+        map.put(688, singletonList("TV"));
+        map.put(689, singletonList("PF"));
+        map.put(690, singletonList("TK"));
+        map.put(691, singletonList("FM"));
+        map.put(692, singletonList("MH"));
+        map.put(800, singletonList("001"));
+        map.put(808, singletonList("001"));
+        map.put(850, singletonList("KP"));
+        map.put(852, singletonList("HK"));
+        map.put(853, singletonList("MO"));
+        map.put(855, singletonList("KH"));
+        map.put(856, singletonList("LA"));
+        map.put(870, singletonList("001"));
+        map.put(878, singletonList("001"));
+        map.put(880, singletonList("BD"));
+        map.put(881, singletonList("001"));
+        map.put(882, singletonList("001"));
+        map.put(883, singletonList("001"));
+        map.put(886, singletonList("TW"));
+        map.put(888, singletonList("001"));
+        map.put(960, singletonList("MV"));
+        map.put(961, singletonList("LB"));
+        map.put(962, singletonList("JO"));
+        map.put(963, singletonList("SY"));
+        map.put(964, singletonList("IQ"));
+        map.put(965, singletonList("KW"));
+        map.put(966, singletonList("SA"));
+        map.put(967, singletonList("YE"));
+        map.put(968, singletonList("OM"));
+        map.put(970, singletonList("PS"));
+        map.put(971, singletonList("AE"));
+        map.put(972, singletonList("IL"));
+        map.put(973, singletonList("BH"));
+        map.put(974, singletonList("QA"));
+        map.put(975, singletonList("BT"));
+        map.put(976, singletonList("MN"));
+        map.put(977, singletonList("NP"));
+        map.put(979, singletonList("001"));
+        map.put(992, singletonList("TJ"));
+        map.put(993, singletonList("TM"));
+        map.put(994, singletonList("AZ"));
+        map.put(995, singletonList("GE"));
+        map.put(996, singletonList("KG"));
+        map.put(998, singletonList("UZ"));
+
+        return map;
+    }
+
+    private static synchronized Map<String, Integer> createCountryCodeByIsoMap() {
+        Map<String, Integer> map = new HashMap<>(MAX_COUNTRIES);
+
+        map.put("AF", 93);
```

Comment:
Could we single source this from the other map?

```java
SparseArray<List<String>> codeToRegion = createCountryCodeToRegionCodeMap();

for(int i = 0; i < codeToRegion.size(); i++) {
   int code = sparseArray.keyAt(i);
   List<Sttring> regions = sparseArray.get(code);

   for (String region : regions) {
      map.put(region, code);
    }
}
```

Reply:
Or maybe it's not the same map in reverse and that was just the case for the few I checked.  In which case, ignore me.

Reply:
@samtstern Thanks, I didn't notice that! 😄

The generated maps weren't exactly the same, but I finally managed to generate a decent diff. The hard coded map has `{ HM=672, GS=500, XK=381 }` additional values over the generated map whereas the latter has this extra value: `TA=290`. Could that actually be a mistake? If I added those 3 extra values to the map full of lists, it would make them perfect inverses. Should I do that?

Reply:
To be safe let's make them perfect inverses and then add those exceptions only where we have them now.  So basically each map is a view on a base map plus some exceptions.

Then we can file an issue to track down the source of those exceptions with Ashwin, but I don't want to block this PR on that investigation.

Reply:
SGTM, is 01d0adb what you were looking for?